### ISSUE TITLE
Setting use_cuda=True hangs and gets killed.

### ISSUE THREAD

**Issue by lorinczszabolcs** [baal-org/baal#234]
**Description of error**
If I set `use_cuda=True` parameter in `wrapped_model.predict_on_dataset()`, the process stops at 0 iterations and gets killed in a short time afterwards. With `use_cuda=False` the process works.

I am using a custom data set, but it doesn't involve anything special, it returns sample, target tuples. Tried with both `replicate_in_memory` set to `True` and `False`, the issue persists with both settings.

The rough code that I am trying to run:

```python
mcdc_predict_model = MCDropoutConnectModule(
                predict_model, layers=["Linear"], weight_dropout=0.9
            )
# using NLL instead of CrossEntropy due to model output log_softmax
wrapped_model = ModelWrapper(
    mcdc_predict_model, torch.nn.NLLLoss(), replicate_in_memory=True
)
predictions = wrapped_model.predict_on_dataset(
    predict_dataset, batch_size=16, iterations=20, use_cuda=False, workers=16
)
```
, where `predict_model` is a `pytorch_lightning.core.LightningModule` object, while `predict_dataset` is a `torch.Dataset` object.

The output I see is:
```
[14664-MainThread] [baal.modelwrapper:predict_on_dataset_generator:263] 2022-09-14T11:40:17.998510Z [info     ]
Start Predict             dataset=40000
0%|                       | 0/2500 [00:00<?, ?it/s]
Killed
```

**Comment by Dref360** [comment]
Hello,

Thank you for submitting the issue, I'll be able to debug this over the weekend. 

What is your configuration in terms of GPUs, Pytorch-Lightning versions and type of model? 

Would it be possible to have a self-contained script I can debug? Maybe using a random model and random data.

**Comment by lorinczszabolcs** [comment]
Hi,

Thanks for your quick reply!

The program is running in a Docker container on WSL2 Ubuntu 20.04, the GPU is a NVIDIA GeForce RTX 3060 Laptop GPU. I actually think I pinpointed the cause, because since reporting the bug I faced the same 'Killed' message several times in different scenarios, all being when I ran out of GPU memory. I think the same happened when I used the `use_cuda` parameter, so the issue could be closed in my opinion, although I was expectin a CUDA: ran out of memory message instead of 'Killed'.

I will soon come up with a script and post it here.
